hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ConstantSizeRegionSplitPolicy.java

```diff
@@ -68,22 +76,14 @@ protected void configureForRegion(HRegion region) {
 
   @Override
   protected boolean shouldSplit() {
-    boolean foundABigStore = false;
-
+    // If any of the stores is unable to split (eg they contain reference files)
+    // then don't split
     for (HStore store : region.getStores()) {
-      // If any of the stores are unable to split (eg they contain reference files)
-      // then don't split
-      if ((!store.canSplit())) {
+      if (!store.canSplit()) {
```

Comment:
I moved the canSplit checking to the front purposely, it has benefits below.
1. Make the logic more clear.
2. If can not split, then we do not need to get tableRegionsCount and sizeToCheck in IncreasingToUpperBoundRegionSplitPolicy, and also avoid other comparsion in loop too.

Reply:
>Make the logic more clear.
If can not split, then we do not need to get tableRegionsCount and sizeToCheck in IncreasingToUpperBoundRegionSplitPolicy, and also avoid other comparsion in loop too.

That's a problem for IncreasingToUpperBoundRegionSplitPolicy. We shouldn't penalise a parent class because of subclasses implementation specifics. And you are already overriding this method on IncreasingToUpperBoundRegionSplitPolicy to do the extra loop anyways, so why keep this extra, needless loop here?

Reply:
>Make the logic more clear.
If can not split, then we do not need to get tableRegionsCount and sizeToCheck in IncreasingToUpperBoundRegionSplitPolicy, and also avoid other comparsion in loop too.

That's a problem for IncreasingToUpperBoundRegionSplitPolicy. We shouldn't penalise a parent class because of subclasses implementation specifics. And you are already overriding this method on IncreasingToUpperBoundRegionSplitPolicy to do the extra loop anyways, so why keep this extra, needless loop here?

Reply:
Although IncreasingToUpperBoundRegionSplitPolicy override the shouldSplit, but does not override isExceedSize.
IMO, even if there is only ConstantSizeRegionSplitPolicy, the ref check should be precondition of size check, since if any store has ref file, all stores are no need to check.

Reply:
@wchevreuil  WDYT?